redislock_test.go

```diff
@@ -142,6 +142,61 @@ var _ = Describe("Client", func() {
 		Expect(numLocks).To(Equal(int32(1)))
 	})
 
+	It("should prevent multiple locks with linear retry (fuzzing)", func() {
+		numLocks := int32(0)
+		wg := new(sync.WaitGroup)
+		var cacheOpts = &redislock.Options{
+			RetryStrategy: redislock.LimitRetry(redislock.LinearBackoff(30*time.Millisecond), 10),
+		}
+		for i := 0; i < 1000; i++ {
+			wg.Add(1)
+
+			go func() {
+				defer GinkgoRecover()
+				defer wg.Done()
+
+				wait := rand.Int63n(int64(50 * time.Millisecond))
+				time.Sleep(time.Duration(wait))
+
+				_, err := subject.Obtain(ctx, lockKey, time.Minute, cacheOpts)
+				if err == redislock.ErrNotObtained {
+					return
+				}
+				Expect(err).NotTo(HaveOccurred())
+				atomic.AddInt32(&numLocks, 1)
+			}()
+		}
+		wg.Wait()
+		Expect(numLocks).To(Equal(int32(1)))
+	})
+
+	It("should prevent multiple locks with exponential retry (fuzzing)", func() {
+		numLocks := int32(0)
+		wg := new(sync.WaitGroup)
+		var cacheOpts = &redislock.Options{
+			RetryStrategy: redislock.LimitRetry(redislock.ExponentialBackoff(30*time.Millisecond, 5*time.Second), 10),
+		}
+		for i := 0; i < 1000; i++ {
+			wg.Add(1)
+
+			go func() {
+				defer GinkgoRecover()
+				defer wg.Done()
+
+				wait := rand.Int63n(int64(50 * time.Millisecond))
```

Comment:
I prefer to remove the wait, it doesn't really change the test, just slows it down

Reply:
It's removed.